src/Faithlife.Tracing.Http/HttpTracingUtility.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+
+namespace Faithlife.Tracing.Http
+{
+	public static class HttpTracingUtility
+	{
+		public static Func<HttpWebRequest, IDisposable> CreateStartTrace(ITraceProvider traceProvider)
+		{
+			return r =>
+			{
+				var currentTrace = traceProvider?.CurrentTrace;
+				if (currentTrace == null)
+				{
+					r.Headers[HttpHeaderNames.B3Sampled] = "0";
+					return null;
+				}
+				else
+				{
+					r.Headers[HttpHeaderNames.B3Sampled] = "1";
+					var requestTrace = currentTrace.Factory.CreateTrace(currentTrace, TraceKind.Client, r.RequestUri.Authority, r.RequestUri.AbsoluteUri, "http.url", r.RequestUri.AbsoluteUri);
+					requestTrace?.Factory.InjectTrace(requestTrace, (k, v) => r.Headers[k] = v);
+					return requestTrace;
+				}
+			};
+		}
+
+		public static HttpClient CreateHttpClient(ITraceProvider traceProvider, HttpMessageHandler messageHandler) => traceProvider == null ? new HttpClient() : new HttpClient(new TracingHttpMessageHandler(traceProvider, messageHandler));
+
+		private sealed class TracingHttpMessageHandler : MessageProcessingHandler
```

Comment:
Might be nice to allow clients to attach this to their own message handler?

Reply:
?

It attaches to the `messageHandler` passed to `CreateHttpClient` above.

Reply:
Oh, expose the created `HttpMessageHandler` for others to delegate to? Yes, that should be the public API.